### PR TITLE
adding iteration count and rotating flag to the bench config

### PR DESCRIPTION
**Pull request by babakpst**
- added iteration count based on the latency of each size,
- added beta = 0 
- added rotating flag

### CODE REVIEW

**Comment by nakajee** [comment]
Can we have some manual or help message about usage (regarding new parameters)?

File: tuning/automation/rocblas-benchInputCreator.py [hunk 1]
```diff
@@ -215,38 +231,140 @@ def createYaml(args, outputfile, problem, sizeMappings, verify):
 
         sizeParams = getSizeParams(size, transA, transB)
 
+        latency = 2*sizeParams['M']*sizeParams['N']*sizeParams['K']/perf[1]/1000 # us
+        niters = math.ceil( 1.15 * 3e6 / latency)
+        
+        otherParams = {"alpha": 1, "beta": 1, "iters": niters, "cold_iters": niters}
+        otherParams_rotating = {"alpha": 1, "beta": 1, "iters": niters, "cold_iters": niters, "flush_memory_size": 536870812}
```
Comment (smalekta) [inline tuning/automation/rocblas-benchInputCreator.py:238]: NDB (no action is needed) const numbers might be better defined on top and not repeated

**Review by smalekta — APPROVED** [review]

**Comment by nakajee** [comment]
> @nakajee There is no change in the running command parameters. Is that what you are asking? I can add a comment to explain the output.

OK. I thought some new options were added to specify new config parameters....

File: tuning/automation/rocblas-benchInputCreator.py [hunk 2]
```diff
@@ -215,38 +231,140 @@ def createYaml(args, outputfile, problem, sizeMappings, verify):
 
         sizeParams = getSizeParams(size, transA, transB)
 
+        latency = 2*sizeParams['M']*sizeParams['N']*sizeParams['K']/perf[1]/1000 # us
```
Comment (nakajee) [inline tuning/automation/rocblas-benchInputCreator.py:234]: Don't we need to consider batch_count?

File: tuning/automation/rocblas-benchInputCreator.py [hunk 3]
```diff
@@ -215,38 +231,140 @@ def createYaml(args, outputfile, problem, sizeMappings, verify):
...
+        niters = math.ceil( 1.15 * 3e6 / latency)
+        
+        otherParams = {"alpha": 1, "beta": 1, "iters": niters, "cold_iters": niters}
+        otherParams_rotating = {"alpha": 1, "beta": 1, "iters": niters, "cold_iters": niters, "flush_memory_size": 536870812}
+        
+        niters=math.ceil( 1.25 * 3e6 / latency)
+        otherParams_beta0 = {"alpha": 1, "beta": 0, "iters": niters, "cold_iters": niters}
```
Comment (nakajee) [inline tuning/automation/rocblas-benchInputCreator.py:241]: Can we have an option to use default iteration (10, 2)?
I usually use this tool before converting Gflops into frequency.

Comment (nakajee) [reply tuning/automation/rocblas-benchInputCreator.py:241]: Or, use default value if perf is >100?

File: tuning/automation/rocblas-benchInputCreator.py (hunk 2 above)
Comment (babakpst) [reply tuning/automation/rocblas-benchInputCreator.py:234]: I'll review the batched GEMM and will update the equation if required.

File: tuning/automation/rocblas-benchInputCreator.py [hunk 4]
```diff
@@ -215,38 +231,140 @@ def createYaml(args, outputfile, problem, sizeMappings, verify):
...
+            yaml.safe_dump(bench_beta0, f, default_flow_style=None, sort_keys=False, width=5000)
+            f.write(f"# End of {name} \n")
+
+        name = outputfile+"_bench_rotating.yaml"
+        benchPath = os.path.join(args.outDir, name)
+        with open(benchPath, "w") as f:
+            yaml.safe_dump(bench_rotating, f, default_flow_style=None, sort_keys=False, width=5000)
+            f.write(f"# End of {name} \n")
+
+        name = outputfile+"_bench_beta0_rotating.yaml"
```
[the remark this answers is not in the document]
Comment (babakpst) [reply tuning/automation/rocblas-benchInputCreator.py]: fixed.

File: tuning/automation/rocblas-benchInputCreator.py (hunk 1 above)
Comment (babakpst) [reply tuning/automation/rocblas-benchInputCreator.py]: done.

File: tuning/automation/rocblas-benchInputCreator.py (hunk 2 above)
Comment (babakpst) [reply tuning/automation/rocblas-benchInputCreator.py]: batch count added.

File: tuning/automation/rocblas-benchInputCreator.py (hunk 3 above)
Comment (babakpst) [reply tuning/automation/rocblas-benchInputCreator.py]: If you run this script as before, it uses the default iters (10/2), but if you indicate -d duration, it updates the iterations.

**Review by nakajee — APPROVED** [review]

**Pull request merged**